### PR TITLE
Move tzif crate from experimental to utils for publishing

### PR DESCRIPTION
**Pull request by nordzilla**

### CODE REVIEW

**Review by Manishearth — APPROVED** [review]
lgtm, after publishing be sure to also run `cargo owner -a github:unicode-org:icu4x-release`

File: utils/tzif/Cargo.toml [hunk 1]
```diff
@@ -9,6 +9,7 @@ version = "0.1.0"
 edition = "2021"
 readme = "README.md"
 license-file = "LICENSE"
+keywords = ["time-zone", "posix", "iana", "parse", "data"]
```
Comment (sffc) [inline utils/tzif/Cargo.toml:13]: Nit: I am getting the following warning. Do you know what the problem is, and can you fix it?

```
warning: /path/to/icu4x/experimental/tzif/Cargo.toml: unused manifest key: package.author
```

Comment (Manishearth) [reply utils/tzif/Cargo.toml:13]: Yeah that key is deprecated now I think

Comment (nordzilla) [reply utils/tzif/Cargo.toml:13]: Ah, this was just a mistake, it should be `authors`. 

Thank you.

File: utils/tzif/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+);
+let data = tzif().parse(stream).unwrap();
+```
+
+#### Parse POSIX time-zone strings
+```rust
+use combine::Parser;
+use tzif::posix_tz_string;
+
+let data = posix_tz_string()
```
[the remark this answers is not in the document]
Comment (nordzilla) [reply utils/tzif/README.md:32]: Configuration options in what sense? 

I don't intend to publish this as `1.0` right away. I'm not sure what configuration would be needed, but if it is we can always add it. 

At this time I'd like to prioritize other ICU4X tasks.

File: utils/tzif/README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+use combine::{Parser, stream};
+use std::fs::File;
+use tzif::tzif;
+
+let file = File::open("path_to_file").unwrap();
+let stream = stream::buffered::Stream::new(
+    stream::position::Stream::new(stream::read::Stream::new(file)),
+    0, /* lookahead */
+);
+let data = tzif().parse(stream).unwrap();
```
[the remark this answers is not in the document]
Comment (nordzilla) [reply utils/tzif/README.md:24]: I'm not an expert with `combine` but this seems to be normal based on their examples:

**Definition**
https://github.com/Marwes/combine/blob/6de4b29d69ccd03c6ab8bef66a4eeaf588e50a92/examples/date.rs#L154-L159

**Invocation**
https://github.com/Marwes/combine/blob/6de4b29d69ccd03c6ab8bef66a4eeaf588e50a92/examples/date.rs#L231-L233

**Review by Manishearth — APPROVED** [review]

**Review by sffc — APPROVED** [review]

**Pull request merged**